### DISCUSSION TITLE
failed to create one more record in 1-1 self relation

### DISCUSSION THREAD

**Discussion by ctsin** [prisma/prisma#21229] [Q&A] [answered]
### Question

With Prisma v5.3.1, and MongoDB provider.

I followed the [official docs](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations#one-to-one-self-relations) about 1-1 self relation, and have a schema: 

> 1-1, 1-m, and m-n self relation receives exactly same issue

```prisma
model SelfOneToOne {
  id          String        @id @default(auto()) @map("_id") @db.ObjectId
  name        String        @unique
  created     DateTime      @default(now())
  superiorID  String?       @unique @db.ObjectId
  superior    SelfOneToOne? @relation("selfRelation", fields: [superiorID], references: [id], onUpdate: NoAction, onDelete: NoAction)
  subordinate SelfOneToOne? @relation("selfRelation")
}
```

Create first record successfully.

```ts
try {
    const user = await prisma.userOfficial.create({
      data: {
        name: "name 1",
        predecessor: {
          create: {
            name: "name predecessor 1",
          },
        },
        successor: {
          create: {
            name: "name successor 1",
          },
        },
      },
    });

    return NextResponse.json(user);
  } catch (error) {
    return NextResponse.json(error);
  }
```

When run the creation with "name 2", Prisma will throw an error P2002.

```ts
try {
    const user = await prisma.userOfficial.create({
      data: {
        name: "name 2",
        predecessor: {
          create: {
            name: "name predecessor 2",
          },
        },
        successor: {
          create: {
            name: "name successor 2",
          },
        },
      },
    });

    return NextResponse.json(user);
  } catch (error) {
    return NextResponse.json(error);
  }
```

```json
{
    "name": "PrismaClientKnownRequestError",
    "code": "P2002",
    "clientVersion": "5.3.0",
    "meta": {
        "target": "UserOfficial_successorId_key"
    }
}
```

```sh
- error PrismaClientKnownRequestError: 
Invalid `prisma.userOfficial.create()` invocation:

Unique constraint failed on the constraint: `UserOfficial_successorId_key`
```

So is there anything wrong with my schema? And how to resolve it? Thanks guys.

**Comment by ludralph** [comment, marked as answer]
Hi @ctsin 👋 

Thank you for raising this question.

There is nothing wrong with your schema. I do think you just hit a bug documented [here](https://github.com/prisma/prisma/issues/20217). There is a suggested workaround documented [here](https://github.com/prisma/prisma/issues/20217#issuecomment-1638596578) and another one in this previous [discussion](https://github.com/prisma/prisma/discussions/18439#discussioncomment-5405001). Do take a look and let me know if it works for you. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
